src/daft-core/src/array/ops/utf8.rs

```diff
@@ -233,20 +286,66 @@ impl Utf8Array {
                         pattern_len,
                     )),
                     Some(self_v) => {
-                        let arrow_result: DaftResult<arrow2::array::Utf8Array<i64>> = pattern_arrow
-                            .iter()
-                            .map(|pat| match pat {
-                                None => Ok(None),
-                                Some(p) => {
-                                    let re = regex::Regex::new(p)?;
-                                    Ok(re.captures(self_v).and_then(|captures| {
-                                        captures.get(index).map(|m| m.as_str())
-                                    }))
-                                }
-                            })
-                            .collect();
+                        let regexes = pattern_arrow.iter().map(|pat| pat.map(regex::Regex::new));
+                        let iter = std::iter::repeat(Some(self_v)).zip(regexes);
+                        regex_extract_first_match(iter, index, self.name())
+                    }
+                }
+            }
+            // Mismatched len case:
+            (self_len, pattern_len) => Err(DaftError::ComputeError(format!(
+                "lhs and rhs have different length arrays: {self_len} vs {pattern_len}"
```

Comment:
Specify that this occurred in `extract`, and also for the `extract_all` case

Reply:
Will do this for the other str kernels as well